source/appModules/calculator.py

```diff
@@ -112,6 +128,16 @@ def event_UIA_notification(self, obj, nextHandler, displayString=None, activityI
 				and doNotAnnounceCalculatorResults
 			):
 				return
+			# #16573: when pasting, calculator sends a truncated UIA displayString
+			# (for example "Display is 1" when it should be "Display is 12.34").
+			# To fix this, we ignore the displayString sent via the UIA notification,
+			# and fetch the value directly from the UI element.
+			for child in resultElement.children:
+				if child.UIAAutomationId in ("CalculatorResults", "CalculatorAlwaysOnTopResults"):
```

Comment:
What would happen if the display string reported by UIA notification is the same as what's shown on screen?

Reply:
I'm not sure I understand that situation you're considering, but I think the fix would still be applicable to that situation. For example, if you type "1+2=", the calculator will still show the correct answer and NVDA will report it correctly as well.  The UIA notification in this scenario would set displayText="Display is 3", and that string is exactly the same as what we would pull from the element itself.  So effectively, if the previous UIA notification was correct, then this does not change any behavior.